src/NodaTime.Demo/NodaTime.Demo.csproj

```diff
@@ -17,4 +17,26 @@
     <PackageReference Include="NUnit3TestAdapter" Version="3.16.1" />
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.4.0" />
   </ItemGroup>
+
+  <ItemGroup Condition="$(TargetFrameworkIdentifier) == '.NETFramework'">
```

Comment:
Now that we're not changing the targets, can we remove all the conditionality here, and just have a single demo source file?

Reply:
I have removed the conditionality in the csproj but the new code (bd89466952f2a308830842bc91b721bbb0ee56c0) is still capable of handling both .NET Framework and .NET Core. Unlike `NodaTime.Test`, the `NodaTime.Demo` project doesn't exercises corner cases of NodaTime and it's not a problem to multi targe .NET Framework and .NET Core. I have tested: running NodaTime.Demo on both .NET Framework and .NET Core (with `<TargetFrameworks>net48;netcoreapp3.1</TargetFrameworks>`) works fine. 

In this particular test (`XmlSchemaDemo`) I think it's interesting to keep the `XmlCodeExporterAssemblyCreator` class and its related `NodaTimeSchemaImporterExtension` that demonstrates how to intercept the NodaTime XML types and convert them to their equivalent C# types.

The interesting parts of this demo are actually the `NodaTimeSchemaImporterExtension` and `NodaTimeNamingProvider` classes.

Reply:
I'll need to check whether the changes to the demo project file break snippet generation (which has previously been quite brittle with respect to getting Roslyn to load it properly). But I can do that after merging.